docs/workflows/create_workflow_block.md

```diff
@@ -0,0 +1,1898 @@
+            return [
+              OutputDefinition(
+                name="images_match", 
+                kind=[BOOLEAN_KIND],
+              ),
+            ]
+    
+        @classmethod
+        def get_execution_engine_compatibility(cls) -> Optional[str]:
+            return ">=1.0.0,<2.0.0"
+    
+        
+    class ImagesSimilarityBlock(WorkflowBlock):
+      
+        def __init__(self):
+            self._sift = cv2.SIFT_create()
+            self._matcher = cv2.FlannBasedMatcher(dict(algorithm=1, trees=5), dict(checks=50))
+          
+        @classmethod
+        def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+            return ImagesSimilarityManifest
+    
+        def run(
+            self,
+            image_1: WorkflowImageData,
+            image_2: WorkflowImageData,
+            similarity_threshold: float,
+        ) -> BlockResult:
+            image_1_gray = cv2.cvtColor(image_1.numpy_image, cv2.COLOR_BGR2GRAY)
+            image_2_gray = cv2.cvtColor(image_2.numpy_image, cv2.COLOR_BGR2GRAY)
+            kp_1, des_1 = self._sift.detectAndCompute(image_1_gray, None)
+            kp_2, des_2 = self._sift.detectAndCompute(image_2_gray, None)
+            matches = self._matcher.knnMatch(des_1, des_2, k=2)
+            good_matches = []
+            for m, n in matches:
+                if m.distance < similarity_threshold * n.distance:
+                    good_matches.append(m)
+            return {
+                "images_match": len(good_matches) > 0,
+            }
+    ```
+
+    * in line `3` we import OpenCV
+
+    * lines `56-58` defines block constructor, thanks to this - state of block 
+    is initialised once and live through consecutive invocation of `run(...)` method - for 
+    instance when Execution Engine runs on consecutive frames of video
+
+    * lines `70-81` provide implementation of block functionality - the details are trully not
+    important regarding Workflows ecosystem, but there are few details you should focus:
+    
+        * lines `70` and `71` make use of `WorkflowImageData` abstraction, showcasing how 
+        `numpy_image` property can be used to get `np.ndarray` from internal representation of images
+        in Workflows. We advise to expole remaining properties of `WorkflowImageData` to discover more.
+
+        * result of workflow block execution, declared in lines `79-81` is in our case just a dictionary 
+        **with the keys being the names of outputs declared in manifest**, in line `44`. Be sure to provide all
+        declared outputs - otherwise Execution Engine will raise error.
+        
+You may ask yourself how it is possible that implemented block accepts batch-oriented workflow input, but do not 
+operate on batches directly. This is due to the fact that the default block behaviour is to run one-by-one against
+all elements of input batches. We will show how to change that in [advanced topics](#advanced-topics) section.
```

Comment:
I'm not sure if this would be the right place, but would be good to bring into user's attention that if `__init__` introduces batch-speciffic state then user probably want to handle batching manually (i.e. we would initialize separate trackers for each element of batch)

Reply:
yes, that's a good idea

Reply:
👍